docs/architecture/adr-060-baseapp++.md

```diff
@@ -0,0 +1,242 @@
+
+##### Pros
+
+- Apps have more fine control over complete tx ordering and inclusion of transactions 
+- Remove reliance on the Tendermint's mempool to do ordering
+
+##### Cons
+
+- Added implemention complexity
+- Duplicating txs in the mempool 
+
+#### Extra Bytes
+
+The extra bytes approach is where the application/node operator could tell tendermint to send a multiple of the max amount of bytes to be included in a block. When `Prepare` proposal is called Tendermint will provide the app with more transactions that can fit in the block, the Application will say which of the transaction to include and which to discard. While this design removes the complexity of defining a mempool in the application, the application does not have complete control over all the transactions in the mempool. 
+
+##### Pros
+
+- App complexity is low
+- Not duplicating mempool txs in the application 
+
+##### Cons
+
+- App has control over a subset of all transactions in the mempool
+
+
+### Process Proposal
+
+At the process proposal phase Tendermint will ask validators to verify the contents of a block. To do this, Tendermint will send the contents of the proposed block to the application. At this stage the application will verify the content of the block in a stateless manner. If the content fails the stateless checks, the application will have the option to tell Tendermint to vote against the block. 
+
+## Decision
+
+> This section describes our response to these forces. It is stated in full sentences, with active voice. "We will ..."
+> {decision body}
+
+## Consequences
+
+> This section describes the resulting context, after applying the decision. All consequences should be listed here, not just the "positive" ones. A particular decision may have positive, negative, and neutral consequences, but all of them affect the team and project in the future.
+
+### Backwards Compatibility
+
+> All ADRs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The ADR must explain how the author proposes to deal with these incompatibilities. ADR submissions without a sufficient backwards compatibility treatise may be rejected outright.
+
+### Positive
+
+{positive consequences}
+
+### Negative
+
+{negative consequences}
+
+### Neutral
+
+{neutral consequences}
+
+## Further Discussions
+
+While an ADR is in the DRAFT or PROPOSED stage, this section should contain a summary of issues to be solved in future iterations (usually referencing comments from a pull-request discussion).
+Later, this section can optionally list ideas or improvements the author or reviewers found during the analysis of this ADR.
+
```

Comment:
Is it worthwhile evaluating some of the requirements for the application anywhere in this document?
https://github.com/tendermint/tendermint/blob/master/spec/abci%2B%2B/abci%2B%2B_app_requirements.md#formal-requirements

I think coherence (Requirement 5) might be the requirement  where it is easy to make a mistake